Source/SwiftLintFramework/Rules/MultipleClosuresWithTrailingClosureRule.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  MultipleClosuresWithTrailingClosureRule.swift
+//  SwiftLint
+//
+//  Created by Erik Strottmann on 8/26/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct MultipleClosuresWithTrailingClosureRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "multiple_closures_with_trailing_closure",
+        name: "Multiple Closures with Trailing Closure",
+        description: "Trailing closure syntax should not be used when passing more than one closure argument.",
+        kind: .style,
+        nonTriggeringExamples: [
+            "foo.map { $0 + 1 }\n",
+            "foo.reduce(0) { $0 + $1 }\n",
+            "if let foo = bar.map({ $0 + 1 }) {\n\n}\n",
+            "foo.something(param1: { $0 }, param2: { $0 + 1 })\n",
+            "UIView.animate(withDuration: 1.0) {\n" +
+            "    someView.alpha = 0.0\n" +
+            "}"
+        ],
+        triggeringExamples: [
+            "↓foo.something(param1: { $0 }) { $0 + 1 }",
+            "↓UIView.animate(withDuration: 1.0, animations: {\n" +
+                "    someView.alpha = 0.0\n" +
+                "}) { _ in\n" +
+                "    someView.removeFromSuperview()\n" +
+            "}"
+        ]
+    )
+
+    public func validate(file: File, kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard let call = Call(file: file, kind: kind, dictionary: dictionary),
+            call.hasTrailingClosure,
+            call.closureArguments.count > 1
+            else {
+                return []
+        }
+
+        return [
+            StyleViolation(ruleDescription: type(of: self).description,
+                           severity: configuration.severity,
+                           location: Location(file: file, byteOffset: call.offset))
```

Comment:
Do you think it would be better if we added the violation to the trailing closure? For example:

```swift
foo.something(param1: { $0 }) ↓{ $0 + 1 }
```

Reply:
Makes sense to me. I had initially put the violation at the start of the function call because that's where it is for `trailing_closure`, but I think the start of the trailing closure is more precise.